packages/playwright/src/AxeCoreWrapper/DefaultAxeCoreWrapper.cs

```diff
@@ -1,108 +1,297 @@
+
+        /// <inheritdoc/>
+        public async Task<AxeResult> RunLegacyOnLocator(ILocator locator, AxeRunOptions? options = null)
+        {
+            await m_axeContentEmbedder.EmbedAxeCoreIntoPage(locator.Page, options?.Iframes);
+
+            string? paramString = JsonConvert.SerializeObject(options);
+            string runParamTemplate = options != null ? "JSON.parse(runOptions)" : string.Empty;
+
+            object jsonObject = await locator.EvaluateAsync<object>($"(node, runOptions) => window.axe.run(node, {runParamTemplate})", paramString);
+            return DeserializeResult(jsonObject);
+        }
+
+        private static async Task<AxeResult> EvaluateAxeRun(IPage page, AxeRunContext? context = null, object? param = null)
+        {
+            string? paramString = JsonConvert.SerializeObject(param);
+
+            string runParamTemplate = param != null ? "JSON.parse(runOptions)" : string.Empty;
+
+            string? contextParam = context is null ? string.Empty : ($"JSON.parse(\'{JsonConvert.SerializeObject(context)}\'),");
+
+            object jsonObject = await page.EvaluateAsync<object>($"(runOptions) => window.axe.run({contextParam}{runParamTemplate})", paramString);
+
+            return DeserializeResult(jsonObject);
+        }
+
+        private static AxeResult DeserializeResult(object jsonObject)
+        {
+            if (jsonObject is null)
+            {
+                throw new Exception();
+            }
+
+            JObject jObject = JObject.FromObject(jsonObject);
+
+            return new AxeResult(jObject);
+        }
+
+        private static List<AxeRuleMetadata> DeserializeRuleMetadata(object jsonObject)
+        {
+            string result = JsonConvert.SerializeObject(jsonObject);
+            var results = JsonConvert.DeserializeObject<List<AxeRuleMetadata>>(result);
+
+            if (results is null)
+            {
+                throw new Exception();
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Runs axe via legacyScan.js at a specific context, which will be passed as-is to Selenium for legacyScan.js to interpret, and
+        /// parses/handles the legacyScan.js output per the current builder options.
+        /// </summary>
+        /// <param name="rawContextArg">The value to pass as-is to legacyScan.js to use as the axe.run "context" argument</param>
+        private async Task<AxeResult> RunInner(IFrame frame, object rawContextArg, AxeRunOptions? runOptions)
+        {
+            await ConfigureAxe(frame);
+
+
+            if (runOptions == null)
+            {
+                runOptions = new AxeRunOptions
+                {
+                    Iframes = true
+                };
+            }
+
+            var runPartialExists = await RunPartialExists(frame);
+
+            if (!runPartialExists)
+            {
+                await m_axeContentEmbedder.EmbedAxeCoreIntoPage(frame.Page, runOptions.Iframes, true);

```

Comment:
This branch looks like it's incomplete - it discards context completely and runs against the top-level page even in the locator-in-inframe case. There should probably be test cases covering those cases.